hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsHttpOperation.java

```diff
@@ -391,6 +398,23 @@ public void processResponse(final byte[] buffer, final int offset, final int len
 
     this.statusDescription = getConnResponseMessage();
 
+    /*
```

Comment:
Wont above getResponseCode() and getResponseMessage also lead to connections being established ? If yes, we should probably return right at the start of the method itself if connectionDisconnectedOnError is true.